zrml/asset-router/src/tests/create.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023-2024 Forecasting Technologies LTD.
+//
+// This file is part of Zeitgeist.
+//
+// Zeitgeist is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at
+// your option) any later version.
+//
+// Zeitgeist is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Zeitgeist. If not, see <https://www.gnu.org/licenses/>.
+
+#![cfg(test)]
+
+use super::*;
+use frame_support::traits::tokens::fungibles::Inspect;
+
+#[test]
+fn create_routes_campaign_assets_correctly() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_ok!(AssetRouter::create(CAMPAIGN_ASSET, ALICE, true, CAMPAIGN_ASSET_MIN_BALANCE));
+        assert!(AssetRouter::asset_exists(CAMPAIGN_ASSET));
+        assert!(!AssetRouter::asset_exists(CUSTOM_ASSET));
+        assert!(!AssetRouter::asset_exists(MARKET_ASSET));
+        assert!(!AssetRouter::asset_exists(CURRENCY));
+    });
+}
+
+#[test]
+fn create_routes_custom_assets_correctly() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_ok!(AssetRouter::create(CUSTOM_ASSET, ALICE, true, CAMPAIGN_ASSET_MIN_BALANCE));
+        assert!(AssetRouter::asset_exists(CUSTOM_ASSET));
+        assert!(!AssetRouter::asset_exists(CAMPAIGN_ASSET));
+        assert!(!AssetRouter::asset_exists(MARKET_ASSET));
+        assert!(!AssetRouter::asset_exists(CURRENCY));
+    });
+}
+
+#[test]
+fn create_routes_market_assets_correctly() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_ok!(AssetRouter::create(MARKET_ASSET, ALICE, true, CAMPAIGN_ASSET_MIN_BALANCE));
+        assert!(AssetRouter::asset_exists(MARKET_ASSET));
+        assert!(!AssetRouter::asset_exists(CAMPAIGN_ASSET));
+        assert!(!AssetRouter::asset_exists(CUSTOM_ASSET));
+        assert!(!AssetRouter::asset_exists(CURRENCY));
+    });
+}
+
+#[test]
+fn create_routes_currencies_correctly() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_ok!(AssetRouter::create(CAMPAIGN_ASSET, ALICE, true, CAMPAIGN_ASSET_MIN_BALANCE));
+        assert_ok!(AssetRouter::create(CUSTOM_ASSET, ALICE, true, CAMPAIGN_ASSET_MIN_BALANCE));
+        assert_ok!(AssetRouter::create(MARKET_ASSET, ALICE, true, CAMPAIGN_ASSET_MIN_BALANCE));
+        assert!(AssetRouter::asset_exists(MARKET_ASSET));
+        assert!(AssetRouter::asset_exists(CAMPAIGN_ASSET));
+        assert!(AssetRouter::asset_exists(CUSTOM_ASSET));
+        assert!(!AssetRouter::asset_exists(CURRENCY));
+    });
+}
```

Comment:
Because of `create_routes_currencies_correctly` I would assume you check for different balances created for the asset classes afterwards instead of just calling `asset_exists`, if you wanna check if routing is not messed up.

Reply:
Those conditions are covered in https://github.com/zeitgeistpm/zeitgeist/blob/0f1374352498c7eb5c75c4b0f1ea6b41ade456bc/zrml/asset-router/src/tests/multi_currency.rs#L51-L111